### PR TITLE
Implement state/validators endpoint

### PR DESCRIPTION
**Pull request by ajsutton**
## PR Description
Implement the `/eth/v1/beacon/states/:state_id/validators` endpoint.

## Fixed Issue(s)
#2759 

## Documentation

- [x] I thought about documentation and added the `documentation` label to this PR if updates are required.

### CODE REVIEW

File: data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetStateValidators.java [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+      ctx.header(Header.CACHE_CONTROL, getMaxAgeForSlot(provider, slot));
+      if (provider.isFinalized(slot)) {
+        handlePossiblyGoneResult(ctx, future);
+      } else {
+        handlePossiblyMissingResult(ctx, future);
+      }
+    } catch (ChainDataUnavailableException ex) {
+      LOG.trace(ex);
+      ctx.status(SC_SERVICE_UNAVAILABLE);
```
Comment (rolfyone) [inline data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetStateValidators.java:113]: you'll need to set the result here too, like below, `BadMessage.serviceUnavailable` was just added, should make that trivial

File: data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetStateValidators.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+    } catch (IllegalArgumentException ex) {
+      LOG.trace(ex);
+      ctx.status(SC_BAD_REQUEST);
+      ctx.result(jsonProvider.objectToJSON(new BadRequest(ex.getMessage())));
```
Comment (rolfyone) [inline data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetStateValidators.java]: nit: may not have existed when you wrote this, but now a `BadRequest.badRequest(jsonProvider, ex.getMessage())` if you want to be consistent with the other comment

File: data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetStateValidators.java (hunk 1 above)
Comment (ajsutton) [reply data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetStateValidators.java:113]: Done. Also updated the single validator version that I based this off.

File: data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetStateValidators.java (hunk 2 above)
Comment (ajsutton) [reply data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetStateValidators.java]: Done. Also updated the single validator version that I based this off.

**Review by rolfyone — APPROVED** [review]
LGTM

**Pull request merged**